examples/codenames/Codenames_RL.ipynb

```diff
@@ -0,0 +1,12542 @@
+        "        for role in (\"spymaster\", \"operative\"):\n",
+        "            player = player_dict[team][role]\n",
+        "            role_key = f\"{team}_{role}\"\n",
+        "            # finish conversation with final message\n",
+        "            events = game_state.format_events_for_player(role_key)\n",
+        "            if events:\n",
+        "                player.conversation.append({\"role\": \"user\", \"content\": events})\n",
+        "            player.conversation.append(\n",
+        "                {\n",
+        "                    \"role\": \"user\",\n",
+        "                    \"content\": f\"Game over. {winner} won. Reason: {reason}\",\n",
+        "                }\n",
+        "            )\n",
+        "\n",
+        "            reward = create_margin_reward(\n",
+        "                player, role_key, game_state, winner, reason\n",
+        "            )\n",
+        "            mc = create_mc(player)\n",
+        "            traj = art.Trajectory(\n",
+        "                messages_and_choices=mc,\n",
+        "                reward=reward,\n",
+        "                metrics={\"reward\": reward, **game_metrics},\n",
+        "            )\n",
+        "            trajectory_dict.setdefault(team, {})[role] = traj\n",
+        "\n",
+        "            # report to OpenPipe for observability (best‑effort)\n",
+        "            try:\n",
+        "                await op_client.report(\n",
+        "                    req_payload={\n",
+        "                        \"model\": f\"{player.model}_{team}\",\n",
+        "                        \"messages\": player.conversation,\n",
+        "                        \"metadata\": {\n",
+        "                            \"game_id\": game_state.game_id,\n",
+        "                            \"rollout_id\": rollout_id,\n",
+        "                            \"iter\": iter_step,\n",
+        "                            \"role\": role,\n",
+        "                            \"team\": team,\n",
+        "                            \"num_turns\": game_state.turn_count,\n",
+        "                            \"blue_words_left\": game_state.words_left[\"blue\"],\n",
+        "                            \"red_words_left\": game_state.words_left[\"red\"],\n",
+        "                            \"winning_color\": winner,\n",
+        "                            \"game_end_reason\": reason[:1000],\n",
+        "                            \"reward\": reward,\n",
+        "                        },\n",
+        "                    },\n",
+        "                    resp_payload={\n",
+        "                        \"id\": f\"chatcmpl-{uuid.uuid4()}\",\n",
+        "                        \"object\": \"chat.completion\",\n",
+        "                        \"created\": int(time.time()),\n",
+        "                        \"model\": \"dummy\",\n",
```

Comment:
Could we pass in whatever model we're testing here? it should be possible to include that as a scenario variable.